### PR TITLE
Bug 1310972 - Use job id as input to log parsing tasks

### PR DESCRIPTION
**Pull request by jgraham**
Now that log information is in the main treeherder database use the id
information there as the input to the log parsing and autoclassification
tasks.

As part of the same change there is a mild refactor of the
autoclassification related tasks to not go through the call_command
infrastructure as this is just unnecessary overhead.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1932)

### CODE REVIEW

File: treeherder/log_parser/utils.py [hunk 1]
```diff
@@ -29,14 +29,12 @@ def extract_text_log_artifacts(project, log_url, job_guid):
     return artifact_list
 
 
-def post_log_artifacts(project, job_guid, job_log):
+def post_log_artifacts(job_log):
     """Post a list of artifacts to a job."""
-    log_url = job_log.url
-    log_description = "%s %s (%s)" % (project, job_guid, log_url)
```
Comment (edmorley) [inline treeherder/log_parser/utils.py]: The log URL is now no longer printed in the case of errors, so the user has to go manually look it up.
This may be fine now we're using New Relic custom attributes (and so relying more on New Relic than the papertrail logs), but just wanted to check.

File: treeherder/log_parser/tasks.py [hunk 2]
```diff
@@ -142,29 +82,28 @@ def parse_job_log(project, func_name, routing_key, job_guid, job_log_id):
...
     if settings.AUTOCLASSIFY_JOBS:
-        autoclassify.apply_async(args=[project, job_guid],
+        autoclassify.apply_async(args=[job_id.job.id],
                                  routing_key="autoclassify.%s" % priority)
 
 
 @retryable_task(name='crossreference-error-lines', max_retries=10)
-def crossreference_error_lines(project, job_guid):
+def crossreference_error_lines(job_id):
     """This task is a wrapper for the crossreference error lines command."""
```
Comment (edmorley) [inline treeherder/log_parser/tasks.py]: This comment is out of date.

Also, looks like the rest of the changes to this function are incomplete? (Nothing has replaced the `call_command` yet)

File: treeherder/log_parser/tasks.py [hunk 3]
```diff
@@ -20,84 +21,26 @@ def parser_task(f):
...
-        return f(project, job_guid, job_log, priority)
+        return f(job_log, priority)
 
     inner.__name__ = f.__name__
     inner.__doc__ = f.__doc__
 
     return inner
 
 
-def parse_job_logs(project, tasks):
```
Comment (edmorley) [inline treeherder/log_parser/tasks.py]: This removal could be a separate commit (within the same PR).

File: tests/autoclassify/test_detect_intermittents.py [hunk 4]
```diff
@@ -26,8 +26,7 @@ def test_detect_intermittents(test_job, failure_lines, classified_failures,
     MatcherManager._detector_funcs = {}
     detector = MatcherManager.register_detector(TestFailureDetector)
 
-    call_command('detect_intermittents', retriggered_job.repository.name,
-                 retriggered_job.guid)
+    call_command('detect_intermittents', str(test_job.id))
```
Comment (edmorley) [inline tests/autoclassify/test_detect_intermittents.py]: Perhaps we should just get rid of the management command entirely?

Or at the least, shouldn't the tests also be switched from `call_command` to triggering a sync celery task too?

File: treeherder/log_parser/utils.py (hunk 1 above)
Comment (jgraham) [inline treeherder/log_parser/utils.py]: I don't think that's a big problem (as you say the same applies to NR, which is almost always the first port of call and by the time you are looking at papertrail, it's likely that you'll have a db session open anyway).

File: tests/autoclassify/test_detect_intermittents.py (hunk 4 above)
Comment (wlach) [inline tests/autoclassify/test_detect_intermittents.py]: I think I mentioned this on irc earlier, but I think the thing to do is move the code in the management command to a python module, then call it as part of the celery task. New relic suggests there's a large amount of overhead to the way we're currently doing things (since we need to start up a whole new python process, create a new database connection, etc). We could keep the management command as well, though honestly I doubt it would ever get used.

File: treeherder/log_parser/crossreference.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    the fact that serialization of mozlog (and hence errorsummary files) is determinisic
+    so we can reserialize each structured error line and perform an in-order textual
+    match.
+
+    :job: - Job for which to perform the crossreferencing
+    """
+    failure_lines = FailureLine.objects.filter(job_guid=job.guid)
+
+    text_log_errors = TextLogError.objects.filter(
+        step__job__guid=job.guid).order_by('line_number')
```
Comment (wlach) [inline treeherder/log_parser/crossreference.py]: You can just do `step__job=job` here.

Incidentally it would be nice if FailureLine and friends referenced the Job model directly via a foreign key relation, but that can obviously wait for another day.

Comment (jgraham) [inline treeherder/log_parser/crossreference.py]: Oh, I have so many db changes coming… but they are all waiting for simpler changes like this to land first :)

File: tests/autoclassify/test_classify_failures.py [hunk 6]
```diff
@@ -34,7 +34,7 @@ def autoclassify(jm, job, test_failure_lines, matchers, status="testfailed"):
 
     register_matchers(*matchers)
 
-    call_command('autoclassify', jm.project, job.guid)
+    call_command('autoclassify', str(job.id))
```
Comment (wlach) [inline tests/autoclassify/test_classify_failures.py]: I'm guessing you're using call_command here to make sure that the commands still work? If so, could you put a comment here (and elsewhere where you're doing this) explaining that this is what's going on?
